cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -100,18 +101,68 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 		errs = append(errs, err)
 	}
 
-	// Upgrade kube-dns/CoreDNS and kube-proxy
-	if err := dns.EnsureDNSAddon(&cfg.ClusterConfiguration, client); err != nil {
-		errs = append(errs, err)
+	// If the coredns / kube-dns ConfigMaps are missing, show a warning and assume that the
+	// DNS addon was skipped during "kubeadm init", and that its redeployment on upgrade is not desired.
+	//
+	// TODO: remove this once "kubeadm upgrade apply" phases are supported:
+	//   https://github.com/kubernetes/kubeadm/issues/1318
+	var missingCoreDNSConfigMap, missingKubeDNSConfigMap bool
+	if _, err := client.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(
+		context.TODO(),
+		kubeadmconstants.CoreDNSConfigMap,
+		metav1.GetOptions{},
+	); err != nil && apierrors.IsNotFound(err) {
```

Comment:
I can remember, that this can also be a "forbidden" error here (due to RBACs not getting created as well). This is true for all config maps here. You might have to use `apierrors.IsForbidden(err)` too.

Reply:
hm, if the CM is forbidden then the kubeadm client has no privileges to access it.
given this is using the existing admin.conf i don't see how it can happen unless the API server is behaving. me and @fabriziopandini saw something similar in kinder, yet not sure why it would happen here.